lib/spack/spack/installer.py

```diff
@@ -1555,7 +1555,7 @@ def _install_action(self, task):
             return InstallAction.INSTALL
 
         # Ensure install_tree projections have not changed.
-        assert task.pkg.prefix == rec.path
+        assert os.path.samefile(task.pkg.prefix, rec.path)
```

Comment:
This is changing a path comparison to be about lower level filesystem details about each file rather than just path equivalency. I assume that this is OK, and at the very least the paths are still being compared so we're fine in that regard, but I just want to double check this is the right move as opposed to using something like `pathlib`.

Reply:
I dug into this a bit more than I did the first time. It looks like os.path.samefile changes the path strings around if they're a bit invalid. That's why the simple path comparison didn't work but samefile did. I added the bugfix to the underlying path problem and reverted this back to the path comparison they had before.